tools/code_format/check_format.py

```diff
@@ -885,6 +885,7 @@ def check_build_line(self, line, file_path, report_error):
                 "//source/common/protobuf instead.")
         if (self.envoy_build_rule_check and not self.is_starlark_file(file_path)
                 and not self.is_workspace_file(file_path)
+                and not self.is_api_file(file_path)
```

Comment:
cc @envoyproxy/dependency-shepherds 

this rule doesnt make sense in the api, which is mounted within an external namespace in envoy ci, and can only refer to the envoy targets through the `@envoy` namespace

perhaps the intention is to prevent accessing the envoy namespace from the api, but then that feels more like a job for `visibility`

Reply:
this could be resolved alternately by shifting these rules into a separate namespace

without this change, the current implementation fails format checking - i guess for good reason